### PR TITLE
add setModifierManager and capabilities methods to @ember/modifier

### PR DESCRIPTION
**Pull request by hmajoros**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/emberjs/ember.js/blob/master/packages/@ember/modifier/index.ts
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. (_for this, it was a missing definition on the existing JS library, so no bump needed_)

### CODE REVIEW

File: types/ember__modifier/index.d.ts [hunk 1]
```diff
@@ -20,3 +20,10 @@ export interface OnModifier extends Opaque<'modifier:on'> {}
  * @see https://api.emberjs.com/ember/4.1/classes/Ember.Templates.helpers/methods/on?anchor=on
  */
 export const on: OnModifier;
+
+export function setModifierManager(
+  factory: (owner: unknown) => unknown,
+  obj: object
+): void;
```
Comment (chriskrycho) [inline types/ember__modifier/index.d.ts:27]: Docs plus types to match [the implementation](https://github.com/glimmerjs/glimmer-vm/blob/4f1bef0d9a8a3c3ebd934c5b6e09de4c5f6e4468/packages/%40glimmer/manager/lib/public/index.ts#L23)!

```suggestion
/**
 * Given a modifier manager factory and a modifier, tell Ember to set the
 * manager returned by that factory as the manager for the modifier.
 *
 * @param factory A function which takes an `owner` and returns a [modifier
 *   manager](https://emberjs.github.io/rfcs/0373-Element-Modifier-Managers.html).
 * @param modifier The modifier definition to associate with the manager.
 */
export function setModifierManager<T>(
  factory: (owner: unknown) => unknown,
  modifier: T
): T;
```

File: types/ember__modifier/index.d.ts [hunk 2]
```diff
@@ -20,3 +20,10 @@ export interface OnModifier extends Opaque<'modifier:on'> {}
...
+
+export function capabilities(version: string): unknown;
```
Comment (chriskrycho) [inline types/ember__modifier/index.d.ts:29]:
```suggestion
/**
 * Given a target version of Ember, return an opaque token which Ember can use
 * to determine what a given modifier manager supports.
 */
export function capabilities(version: string): unknown;
```

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Thanks for this! Added some suggested docs and one small type signature tweak!

**Review by chriskrycho — APPROVED** [review]
Thank you! If you wouldn’t mind opening a PR today or Monday to the corresponding package on the Ember.js repo, that’d be very helpful (it’ll keep the two aligned, important as we prep to move *out* of DT).

**Comment by hmajoros** [comment]
Ready to merge

**Pull request merged**